src/common/transformations/src/transformations/op_conversions/convert_batch_to_space.cpp

```diff
@@ -4,224 +4,216 @@
+        flat_node = rg.make<Reshape>(flat_node, squeezed_shape, special_zero);
 
         //    Crop the start and end of dimensions according to `crops_begin`, `crops_end` to produce
         //    the output of shape:
         //    note: `crops_begin[0], crops_end[0]` are expected to be 0.
         //    `y = [batch / (B_1 * ... * B_{N - 1}), crop(D_1 * B_1, crops_begin[1], crops_end[1]),
         //          crop(D_2 * B_2, crops_begin[2], crops_end[2]), ... ,
         //          crop(D_{N - 1} * B_{N - 1}, crops_begin[N - 1], crops_end[N - 1])]`
-        std::vector<int64_t> upperbounds_values;
-        auto flat_node_shape = flat_node->get_shape();
-        for (size_t i = 0; i < flat_node_shape.size(); ++i) {
-            upperbounds_values.push_back(flat_node_shape.at(i) - crops_end_values.at(i));
-        }
-
-        const auto upperbounds = opset3::Constant::create(crops_end.get_element_type(),
-                                                          Shape{upperbounds_values.size()},
-                                                          upperbounds_values);
+        const auto shape_of_flat_node = rg.make<ShapeOf>(flat_node, crops_end.get_element_type());
+        const auto upperbounds = rg.make<Subtract>(shape_of_flat_node, crops_end);
 
-        std::vector<int64_t> begin_mask(data_shape.size(), 0);
-        std::vector<int64_t> end_mask(data_shape.size(), 0);
-        flat_node =
-            std::make_shared<opset3::StridedSlice>(flat_node, crops_begin_const, upperbounds, begin_mask, end_mask);
-        new_ops.push_back(flat_node);
+        const auto begin_mask = vector<int64_t>(data_shape_rank.get_length(), 0);
+        const auto& end_mask = begin_mask;
+        flat_node = rg.make<StridedSlice>(flat_node, crops_begin, upperbounds, begin_mask, end_mask);
 
         flat_node->set_friendly_name(batch_to_space->get_friendly_name());
-        ngraph::copy_runtime_info(batch_to_space, new_ops);
-        ngraph::replace_node(batch_to_space, flat_node);
+        copy_runtime_info(batch_to_space, rg.get());
+        replace_node(batch_to_space, flat_node);
         return true;
     };
 
-    auto m = std::make_shared<ngraph::pattern::Matcher>(batch_to_space, matcher_name);
+    const auto m = make_shared<pattern::Matcher>(batch_to_space, matcher_name);
     this->register_matcher(m, callback);
 }
 
 void ov::pass::ConvertBatchToSpace::convert_batch_to_space_by_elements() {
     MATCHER_SCOPE(ConvertBatchToSpace_convert_batch_to_space_by_elements);
-    auto batch_to_space = ngraph::pattern::wrap_type<ov::opset3::BatchToSpace>();
+    const auto batch_to_space = pattern::wrap_type<BatchToSpace>();
     matcher_pass_callback callback = [this](pattern::Matcher& m) {
-        auto batch_to_space = std::dynamic_pointer_cast<ov::opset3::BatchToSpace>(m.get_match_root());
-        if (!batch_to_space) {
+        const auto batch_to_space = dynamic_pointer_cast<BatchToSpace>(m.get_match_root());
```

Comment:
optional: I guess we can move the common parts of the transformations in the separate function

Reply:
I was thinking of that and imo it doesn't fits here as common parts regard local variables instantiation or flow control. So it would reduce code readability.